src/datastore/database.cpp

```diff
@@ -0,0 +1,223 @@
+{
+    /* Get all team and stadium data, create a temp vector that will be used to store souvenir data */
+    std::vector<Team> teamVect(Database::getTeamsVector());
+    std::vector<Stadium> stadiumVect(Database::getStadiumsVector());
+    std::vector<Souvenir> tempSouvenirVect;
+
+    // Initializing - vector will hold all lines of data within the document
+    std::vector<std::vector<std::string>> allRows;
+
+    // Temp Enum variables used to hold string of enum
+    std::string tempLeague;
+    std::string tempRoof;
+    std::string tempSurface;
+    std::string tempTypology;
+
+    /* rowHeader hold the header of a save file */
+    std::vector<std::string> rowHeader;
+    rowHeader.push_back("# id, team name, hidden bool, id, stadium name, capacity, location, playing surface");
+    rowHeader.push_back(" league, date opened, distance to center field, ballpark typology, Rooftype");
+    allRows.push_back(rowHeader);
+
+    for(unsigned int i = 0; i < teamVect.size(); ++i)
+    {
+        // Will hold all the columns within a given row
+        std::vector<std::string> columns;
+
+        /* Convert enum values to their string enum version */
+        tempLeague = Team::LEAGUE_STRING[teamVect[i].league];
+        tempRoof = Stadium::ROOF_STRING[stadiumVect[i].roof];
+        tempSurface = Stadium::SURFACE_STRING[stadiumVect[i].surface];
+        tempTypology = Stadium::TYPOLOGY_STRING[stadiumVect[i].typology];
+
+        /* Adds all the data that belongs to team and it's stadium to the columns vector */
+        columns.push_back(std::to_string(teamVect[i].getId()));
+        columns.push_back(teamVect[i].getName());
+        columns.push_back(std::to_string(teamVect[i].hidden));
+        columns.push_back(std::to_string(teamVect[i].getStadiumId()));
+        columns.push_back(stadiumVect[i].getName());
+        columns.push_back(std::to_string(stadiumVect[i].getSeatCap()));
+        columns.push_back("\""+stadiumVect[i].getLocation()+"\"");
+        columns.push_back(tempSurface);
+        columns.push_back(tempLeague);
+        columns.push_back(std::to_string(stadiumVect[i].getYearOpened()));
+        columns.push_back(std::to_string(stadiumVect[i].getCenterFieldDist()));
+        columns.push_back(tempTypology);
+        columns.push_back(tempRoof);
+
+        // Get's a vector of the teams souvenirs
+        tempSouvenirVect = stadiumVect[i].getSouvenirs();
+        // Number of souvenirs
+        columns.push_back(std::to_string(tempSouvenirVect.size()));
+
+        for(const Souvenir& item: tempSouvenirVect)
+        {
+            /* add all the teams souvenir data to the columns vector */
+            columns.push_back(std::to_string(item.hidden));
+            columns.push_back(item.getName());
+            std::string strPrice(std::to_string(item.getPrice()));
+            // Erasing trailing zeros
+            strPrice.erase(strPrice.find_last_not_of('0') + 1, std::string::npos);
```

Comment:
Are you assuming that the price is $`3.1400000...`? This won't always be true with floating points. The price could actually be $`3.14000...0001`. I suggest you do something similar like this example I found.
```cpp
double pi = 3.14159265359;
stringstream stream;
stream << fixed << setprecision(2) << pi;
string s = stream.str();

Reply:
I'm putting the string onto a vector, not outputting anything here. Just cutting off the trailing zeros after the type doubles are converted to strings

Reply:
I understand what you're trying to do, but I'm saying it won't always work.
![Capture](https://user-images.githubusercontent.com/5513122/56063181-69b1a880-5d23-11e9-8b7e-6baabefaaa58.PNG)

Reply:
You will end up outputting `3.001` to the file, but what you should want is `3.00`

Reply:
Works fine, matches original csv, not adding anything